toplevel/role/role.go

```diff
@@ -59,22 +59,43 @@ func (e entry) Equals(i interface{}) bool {
 		vault.OptionsEqual(e.Options, entry.Options)
 }
 
+// this is what appears in the logs when a role is written to vault
+// issue: keeps running reconcile even when no changes are made
 func (e entry) Save() error {
 	path := filepath.Join("auth", e.Mount.Path, "role", e.Name)
 	options := make(map[string]interface{})
 	for k, v := range e.Options {
-		// local_secret_ids can not be changed after creation so we skip this option
+		// Skip local_secret_ids as it cannot be changed after creation
 		if k == "local_secret_ids" {
 			continue
-			// initially unmarshalled as string or nil and require further processing
 		} else {
 			options[k] = v
 		}
 	}
-	err := vault.WriteSecret(e.Instance.Address, path, vault.KV_V1, options)
+
+	// Read the existing role data from Vault
+	existingData, err := vault.ReadSecret(e.Instance.Address, path, vault.KV_V1)
```

Comment:
Save method should be pure, should not read or check validity, that logic should happen in the save caller or elsewhere